hoomd/hpmc/compute.py

```diff
@@ -102,3 +103,109 @@ def free_volume(self):
+    :math:`x`. For each particle, it finds the smallest scale factor :math:`1+x`
+    that would cause the particle to touch one of its neighbors and records that
+    in the histogram :math:`s(x)`. The histogram is discrete and :math:`s(x_i) =
+    s[i]` where :math:`x_i = i \cdot dx + dx/2`.
+
+    In an NVT simulation, the extrapolation of :math:`s(x)` to :math:`x = 0`,
+    :math:`s(0+)` is related to the pressure.
+
+    .. math::
+        \frac{P}{kT} = \rho \left(1 + \frac{s(0+)}{2d} \right)
+
+    where :math:`d` is the dimensionality of the system and :math:`\rho` is the
+    number density.
+
+    Suggested parameters for most systems (assuming particle diameters are ~1):
+
+      * *xmax* = 0.02
+      * *dx* = 1e-4
+
+    In systems near densest packings, ``dx=1e-5`` may be needed along with
+    either a smaller xmax or a smaller region to fit. A good rule of is to fit
+    a region where ``numpy.sum(s[0:n]*dx)`` ~ 0.5.
+
+    `betaP` performs the curve fit and computes :math:`\frac{P}{kT}` using a
+    polynomial curve fit of degree 5.
+
+    Warning:
+        `SDF` does not compute correct pressures for simulations with
+        concave particles or enthalpic interactions.
+
+    Note:
+
+        `SDF` runs on the CPU even in GPU simulations.
+
+    Examples::
+
+        sdf = hoomd.hpmc.compute.SDF(xmax=0.02, dx=1e-4)
+
+
+    Attributes:
+        xmax (float): Maximum *x* value at the right hand side of the rightmost
+          bin.
+
+        dx (float): Bin width.
+
+    """
+
+    def __init__(self, xmax, dx):
+        # store metadata
+        param_dict = ParameterDict(xmax=float(xmax), dx=float(dx))
+        self._param_dict.update(param_dict)
+
+    def _attach(self):
+        integrator = self._simulation.operations.integrator
+        if not isinstance(integrator, integrate.HPMCIntegrator):
+            raise RuntimeError("The integrator must be an HPMC integrator.")
+
+        # Extract 'Shape' from '<hoomd.hpmc.integrate.Shape object>'
+        integrator_name = integrator.__class__.__name__
+
+        cpp_cls = getattr(_hpmc, 'ComputeSDF' + integrator_name)
+
+        self._cpp_obj = cpp_cls(self._simulation.state._cpp_sys_def,
+                                integrator._cpp_obj, self.xmax, self.dx)
+
+        super()._attach()
+
+    @log(requires_run=True)
+    def sdf(self):
+        """Scale distribution function."""
+        self._cpp_obj.compute(self._simulation.timestep)
+        return self._cpp_obj.sdf
+
+    @log(requires_run=True)
+    def betaP(self):
+        r"""Pressure in NVT simulations.
+
+        Use a polynomial curve fit to estimate :math:`\frac{s(0+)}{2d}` and
+        compute the pressure via:
+        .. math::
+            \frac{P}{kT} = \rho \left(1 + \frac{s(0+)}{2d} \right)
+        """
+        # get the values to fit
+        n_fit = int(numpy.ceil(self.xmax / self.dx))
+        sdf_fit = numpy.asarray(self.sdf[0:n_fit]) / sum(self.sdf[0:n_fit])
```

Comment:
```suggestion
        sdf_fit = self.sdf[0:n_fit]
```
Dividing by the sum is not the appropriate normalization. See the comments in `ComputeSDF.h` on how to correct `self.sdf` to be a probability density.